### DISCUSSION TITLE
CORS issue with Livewire on a subdomain

### DISCUSSION THREAD

**Discussion by barrydevt** [livewire/livewire#5159] [1. Help] [answered]
I have one Laravel installation which has some routes that use a subdomain, eg mysite.com and vendor.mysite.com.

I have an issue where Livewire is not working on any of the subdomain pages due to the usual CORS issues. 

I have tried implementing the correct CORS response headers in Nginx and also using Laravel middleware, but I have had no luck. The Livewrire request always returns 419 and says the access control headers are not present. (Although I can confirm they are every other request to the server).

I got things working by changing some code in `LivewireManager.php` to detect if the url was in the subdomain context and if so, to change the assets url to point to the subdomain, therefore never making any request to the main domain, but it's a hack. 

 Is there any official advice on how to use Livewire on subdomain/multi-tenant sites? I noticed https://github.com/livewire/livewire/discussions/1829 had the same issue, however their 'fix' of changing the asset url in the config did not work for me.

**Comment by joshhanley** [comment]
@barrydevt ah what a pain!

Hmm I'm surprised using the `assert_url` didn't work.

I don't have any offical advice as I've not used Livewire in that setup before.

Just to clarify, do you have the assets on a subdomain or something else, like a tenant domain?

**Reply by joshhanley** [reply]
@barrydevt ah yep, glad you got it sorted! 🙂 Can you add a comment to that discussion with the details, in case anyone else comes across it?

Also can you copy your comment to a new answer below and mark this answered?

Thanks!

**Comment by barrydevt** [comment, marked as answer]
My bad, it appears changing the asset_url in the Livewire config does fix this. I was confused because I was setting

`'asset_url' => '/'`

but it should just be `'asset_url' => ''`

https://github.com/livewire/livewire/discussions/1829 says to use 'asset_url' => env('/'), but obviously there is no environment value for / so it translates to ''
